index.html

```diff
@@ -1320,190 +1320,247 @@ <h1>Core Properties</h1>
+      </p>
+      <p class="note" title="Property names used in multiple classes">
+  The property names <code>id</code>, <code>type</code> and
+  <code>controller</code> can be present in different classes with possible
+  differences in constraints.
+      </p>
 
       <section>
-        <h2>Core Properties for a <a>DID Document</a></h2>
+        <h3>DID Document properties</h3>
 
         <table class=simple>
           <thead>
             <tr>
               <th>Property</th>
-              <th>Usage constraints</th>
+              <th>Required?</th>
               <th>Value constraints</th>
             </tr>
           </thead>
           <tbody>
             <tr>
-              <td><code>id</code></td>
-              <td>REQUIRED</td>
+              <td><code><a>id</a></code></td>
+              <td>yes</td>
               <td>
-                A <a data-cite="INFRA#string">string</a> that conforms to the rules in Section <a href="#did-syntax"></a>.
+A <a data-cite="INFRA#string">string</a> that conforms to the rules in Section
+<a href="#did-syntax"></a>.
               </td>
             </tr>
             <tr>
               <td><code><a>alsoKnownAs</a></code></td>
-              <td>OPTIONAL</td>
+              <td>no</td>
               <td>
-                An <a data-cite="INFRA#ordered-set">ordered set</a> of <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>.
+An <a data-cite="INFRA#ordered-set">ordered set</a> of
+<a data-cite="INFRA#string">strings</a> that conform to the rules of
+[[RFC3986]] for <a>URIs</a>.
               </td>
             </tr>
             <tr>
               <td><code><a>controller</a></code></td>
-              <td>OPTIONAL</td>
+              <td>no</td>
               <td>
-                A <a data-cite="INFRA#string">string</a> or an <a data-cite="INFRA#ordered-set">ordered set</a> of <a data-cite="INFRA#string">strings</a> that conform to the rules in Section <a href="#did-syntax"></a>.
+A <a data-cite="INFRA#string">string</a> or an
+<a data-cite="INFRA#ordered-set">ordered set</a> of
+<a data-cite="INFRA#string">strings</a> that conform to the rules in Section
+<a href="#did-syntax"></a>.
               </td>
             </tr>
             <tr>
               <td><code><a>verificationMethod</a></code></td>
-              <td>OPTIONAL</td>
-              <td>
-                An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification Method</a> <a data-cite="INFRA#ordered-map">maps</a> (see <a href="#core-properties-for-a-verification-method"></a>) or <a data-cite="INFRA#string">strings</a> that conform to the rules of [[RFC3986]] for <a>URIs</a>. <br/>The <a>URI</a> values SHOULD also conform to the rules in Section <a href="#did-url-syntax"></a>.
+              <td>no</td>
+              <td rowspan="6">
+An <a data-cite="INFRA#ordered-set">ordered set</a> of either <a>Verification
```

Comment:
The core text §5.4 says "If present, the value MUST be an ordered set of verification methods, where each verification method is described by a map. ". The possibility to use, instead, a DID-URL is only allowed for the other properties like `keyAgreement` and the like.

(The table was wrong before, because I did not realize this subtle difference. This means that JSON Schema must be updated, too, to reflect this difference. This difference cannot be expressed in RDFS (the verification methods themselves always have a URI, ie, no blank nodes are allowed) and I will have to check the SHACL and SHEX cases.)

Reply:
I wasn't 100% sure about this, so I'm glad you picked up on it. I want to double check with the editors that values of  `verificationMethod` can never contain only a reference and must always be embedded as you say, in which case I'll fix the table. However, the algorithm for embedded vs referenced verification method objects that has been present since the `publicKey` days implies that the value of `verificationMethod` could potentially reference a verification method elsewhere, eg. in another DID document. In which case the table stays the same and the normative language in the `verificationMethod` definition needs updating. But it is ambiguous.

Reply:
From a purely JSON-LD point of view (I know that should not necessarily influence the model) there is no way of checking this, unless the pure JSON structure is also imposed. So I would prefer to allow the usage of a URL as part of the possible values of a `verificationMethod` to avoid this discrepancy on the JSON-LD level.

Reply:
Confirmed with @msporny that the value of `verificationMethod` should NOT be a URL, only a fully qualified verification method, so I will update the table accordingly.

Reply:
@msporny, are you sure?

Consider these two JSON-LD fragments:

- https://tinyurl.com/y4xdj83l (on JSON-LD Playground): taking an example from the spec
- https://tinyurl.com/yyavt36k (on JSON-LD Playground): same example but reformulated in a (clearly uglier) JSON-LD

From a JSON-LD point of view, _these two JSON-LD fragments are identical_, as you can see the N-Quads versions which are literally identical. This means that the JSON-LD serialization of the model is restricted to a particular "profile" of JSON-LD (which, I guess, can be expressed through a JSON Schema). 

If there is a good reason for this I will not object. But let us make it sure that we go into something not really clean with an open eye.

cc @dlongley

Reply:
On more thought on https://github.com/w3c/did-core/pull/571#discussion_r563848624 (almost a day later:-): 

I see now that the text says, in [§6.3.1](https://w3c.github.io/did-core/#production-0) that

> The DID document MUST be serialized according to the production rules for JSON, with one additional requirement: The DID document MUST include the @context property.

which means, indeed, that the [second playground example](https://tinyurl.com/yyavt36k) would not be an acceptable DID Document in JSON-LD. I still do not understand why this restriction is necessary, but I accept that the spec is at least consistent.

But then, I believe, we should add a note that emphasizes that the conformance of the DID Document, when serialized in JSON-LD, is _not_ based on the semantic mapping of JSON-LD on Linked Data. From a Linked Data point of view, making this clear is important.

I also wonder whether it is possible to publish a [JSON-LD Frame document](https://www.w3.org/TR/json-ld11-framing/) that could be used to take any arbitrary, but semantically equivalent JSON-LD (like the aforementioned second playground example). We could then also say that any JSON-LD document, that expresses a DID Document, when transformed using that Frame, should abide to the constraints in the spec. I have no experience with JSON-LD Framing, but we may get some help from some people who have (@dlongley? @davidlehn?)

Reply:
The constraints about top level `id` and and internally nested verification method objects apply to the abstract data model, so I would argue that serializing them to JSON-LD that doesn't exactly match that description but that is functionally the same (as in the second playground example) would be acceptable. If we were hoping to convert between plain JSON and JSON-LD directly it might be a problem, but since we're supposed to go via the ADM in the middle, maybe that's okay? The main thing with the nested verification methods is that you shouldn't be going out to another document to fetch them. In the case of the second playground example, that's still fine.